File: Language/Java/Algorithms/SearchingAlgorithms/BinarySearch/BinarySearch.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

 package Language.Java.Algorithms.SearchingAlgorithms.BinarySearch;
/**
 *
 * @author ryanw
 */
public class BinarySearch {
    public static int search(int[] arr, int target){
        int left = 0;
        int right = arr.length - 1;
        
        while(left <= right){
            int mid = Math.floorDiv(left+right, 2);
            if(arr[mid] == target){
                return mid;
            } 
            if(arr[mid] < right){
                right = mid - 1;
            }
            if(arr[mid] > right){
                left = mid + 1;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        
    }

    
}

File: Language/Python/Algorithms/SearchingAlgorithms/BinarySearch.py
def search(arr, target):
    left, right = 0, len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if(arr[mid] == target):
            return mid
        elif(arr[mid] < target):
            left = mid + 1
        else:
            right = mid - 1
    return -1

# example
array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
target = 5
print(search(array, target)) # Output : 4
print(search(array, 11)) # Output : -1
